src/main/java/com/zendesk/maxwell/RunLoopProcess.java

```diff
@@ -31,8 +34,8 @@ public boolean runLoop() throws Exception {
 			while (this.runState == RunState.RUNNING)
 				work();
 
-			this.beforeStop();
 		} finally {
+			this.beforeStop();
```

Comment:
yeah that's totally true, but it's less harmful than current behavior. 

 I did this to make sure that if `work()` throws we still call `beforeStop` -- we weren't shutting down open-replicator and were leaking threads.

If ultra-paranoid could do:

```
try {
   ...
} finally { 
   try { 
   } finally { 
      this.runState = RunState.STOPPED
```